apps/api/src/controllers/v1/webhooks-v1.controller.ts

```diff
@@ -0,0 +1,84 @@
+import { ApiService } from '#api/api.service';
+import { WatchGraphsDto } from '#lib';
+import { Controller, HttpCode, HttpStatus, Logger, Body, Put, Res, HttpException, Get, Query, Delete, Param } from '@nestjs/common';
+import { ApiBody, ApiOkResponse, ApiOperation, ApiParam, ApiQuery, ApiTags } from '@nestjs/swagger';
+import { Response } from 'express';
+
+@Controller('v1/webhooks')
+@ApiTags('v1/webhooks')
+export class WebhooksControllerV1 {
+  private readonly logger: Logger;
+
+  constructor(private apiService: ApiService) {
+    this.logger = new Logger(this.constructor.name);
+  }
+
+  @Get()
+  @HttpCode(HttpStatus.OK)
+  @ApiOperation({ summary: 'Get all registered webhooks' })
+  @ApiOkResponse({ description: 'Retrieved all registered webhooks' })
+  async getAllWebhooks(): Promise<unknown> {
+    return this.apiService.getAllWebhooks();
+  }
+
+  @Get('users/:msaId')
+  @HttpCode(HttpStatus.OK)
+  @ApiOperation({ summary: 'Get all registered webhooks for a specific MSA ID' })
+  @ApiOkResponse({ description: 'Retrieved all registered webhooks for the given MSA ID' })
+  async getWebhooksForMsa(@Param('msaId') msaId: string): Promise<string[]> {
+    return this.apiService.getWebhooksForMsa(msaId);
+  }
+
+  @Get('urls')
+  @HttpCode(HttpStatus.OK)
+  @ApiOperation({ summary: 'Get all webhooks registered to the specified URL' })
+  @ApiOkResponse({ description: 'Retrieved all webhooks registered to the specified URL' })
+  async getWebhooksForUrl(@Query('url') url: string): Promise<string[]> {
+    return this.apiService.getWatchedGraphsForUrl(url);
+  }
+
+  @Put()
+  @ApiOperation({ summary: 'Watch graphs for specified dsnpIds and receive updates' })
```

Comment:
@JoeCap08055 We were talking a bit about potential missing endpoints. I think a potential missing endpoint here is one to register a webhook for ALL users.

So specific to the renaming, if we were to create such an endpoint, would we modify this endpoint structure in anyway? (Or just change the Dto to have that array optional to mean "all"?)

Reply:
I was thinking the same thing; I think we can just add an interpretation to the DTO for the existing endpoint, such that a zero MSA ID means "register for all users". Or if that seems too hacky, we could modify the DTO in some other way, but make it additive/backward-compatible. No need for a separate, dedicated endpint IMO.